mlflow/tracing/trace_manager.py

```diff
@@ -63,64 +61,21 @@ def __init__(self):
         self._trace_id_to_request_id = {}
         self._lock = threading.Lock()  # Lock for _traces
 
-    def start_detached_span(
-        self,
-        name: str,
-        request_id: Optional[str] = None,
-        parent_id: Optional[str] = None,
-        span_type: str = SpanType.UNKNOWN,
-    ) -> MlflowSpanWrapper:
+    def register_trace(self, trace_id: str, trace_info: TraceInfo):
         """
-        Start a new OpenTelemetry span that is not part of the current trace context, but with the
-        explicit parent span ID if provided.
+        Register a new trace info object to the in-memory trace registry.
 
         Args:
-            name: The name of the span.
-            request_id: The request (trace) ID for the span.
-            parent_id: The parent span ID of the span. If None, the span will be a root span.
-            span_type: The type of the span.
-
-        Returns:
-            The newly created span (wrapped in MlflowSpanWrapper). If any error occurs, returns a
-            NoOpMlflowSpanWrapper that has exact same interface but no-op implementations.
+            trace_id: The trace ID for the new trace.
+            trace_info: The trace info object to be stored.
         """
-        from mlflow.tracing.provider import get_tracer
-
-        try:
-            tracer = get_tracer(__name__)
-            if parent_id:
-                if not request_id:
-                    raise MlflowException(
-                        "Parent span ID is provided without its request ID. Please specify the "
-                        "request ID of the parent span to start a detached span.",
-                        error_code=INVALID_PARAMETER_VALUE,
-                    )
-                parent_span = self.get_span_from_id(request_id, parent_id)
-                if parent_span is None:
-                    raise MlflowException(
-                        f"Parent span with ID '{parent_id}' not found.",
-                        error_code=RESOURCE_DOES_NOT_EXIST,
-                    )
-                context = trace_api.set_span_in_context(parent_span._span)
-            else:
-                context = None
-
-            otel_span = tracer.start_span(name, context=context)
-            if not request_id:
-                request_id = self.get_or_create_request_id(otel_span.get_span_context().trace_id)
-
-            span = MlflowSpanWrapper(otel_span, request_id=request_id, span_type=span_type)
-            self.add_or_update_span(span)
-            return span
-        except Exception as e:
-            _logger.warning(f"Failed to start span {name}: {e}")
-            return NoOpMlflowSpanWrapper()
-
-    def add_or_update_span(self, span: MlflowSpanWrapper):
+        with self._lock:
+            self._traces[trace_info.request_id] = _Trace(trace_info)
+            self._trace_id_to_request_id[trace_id] = trace_info.request_id
+
+    def register_span(self, span: MlflowSpanWrapper):
```

Comment:
note: I realized that we no longer call the `add_or_update_span` except the initial span creation, so it's renamed to `register_span` without `update`.